src/main/java/de/retest/recheck/cli/subcommands/Diff.java

```diff
@@ -1,73 +1,169 @@
+import de.retest.recheck.execution.RecheckDifferenceFinder;
 import de.retest.recheck.ignore.Filter;
-import de.retest.recheck.printer.TestReportPrinter;
+import de.retest.recheck.persistence.GoldenMasterProviderImpl;
+import de.retest.recheck.persistence.PersistenceFactory;
+import de.retest.recheck.printer.ActionReplayResultPrinter;
+import de.retest.recheck.report.ActionReplayResult;
+import de.retest.recheck.report.SuiteReplayResult;
+import de.retest.recheck.report.TestReplayResult;
 import de.retest.recheck.report.TestReport;
 import de.retest.recheck.report.TestReportFilter;
+import de.retest.recheck.ui.descriptors.SutState;
+import picocli.CommandLine;
 import picocli.CommandLine.Command;
 import picocli.CommandLine.Option;
 import picocli.CommandLine.Parameters;
 
-@Command( name = "diff", description = "Display differences of given test report." )
+@Command( name = "diff", description = "Display differences of given Golden Masters." )
 public class Diff implements Runnable {
 
 	private static final Logger logger = LoggerFactory.getLogger( Diff.class );
 
-	@Option( names = "--help", usageHelp = true, hidden = true )
-	private boolean displayHelp;
+	private static final PersistenceFactory persistenceFactory = new PersistenceFactory( getXmlTransformer() );
+	private static final GoldenMasterProviderImpl goldenMasterProvider =
+			new GoldenMasterProviderImpl( persistenceFactory.getPersistence() );
+
+	private static final String TEST_NAME = "Test - Golden Master Comparison";
+	private static final String SUITE_NAME = "Suite - Golden Master Comparison";
+
+	// TODO Temporarily it will be allowed to use the diff command for display of test reports
+	// TODO Will be removed in following version; instead, use the command "show"
+	@Parameters( arity = "1..2", description = GoldenMasterUtil.GOLDEN_MASTER_PARAMETER_DESCRIPTION )
+	private Path[] goldenMasterPath;
 
 	@Option( names = "--exclude", description = FilterUtil.EXCLUDE_OPTION_DESCRIPTION )
 	private List<String> exclude;
 
-	@Parameters( arity = "1", description = TestReportUtil.TEST_REPORT_PARAMETER_DESCRIPTION )
-	private File testReport;
+	@Option( names = "--help", usageHelp = true, hidden = true )
+	private boolean displayHelp;
 
-	@Override
-	public void run() {
-		if ( !PreCondition.isSatisfied() ) {
+	@Option( names = "--output", arity = "1",
+			description = "Save differences of Golden Masters to specified directory as test report." )
+	private Path directory;
+
+	@Override public void run() {
+		// until the following version, a user may still use the diff command to print test report differences.
+		// remove in next version
+		if ( goldenMasterPath.length == 1 ) {
+			logger.warn(
+					"The diff command for viewing a test report is deprecated and will be removed in the following version."
+							+ " Please use the new command 'show' to display a test report." );
+			final String[] args = { goldenMasterPath[0].toString() };
```

Comment:
We should pass as many given arguments as possible to the other command. That includes `excludedFilters`, `help`?, ...

Reply:
Thank you, good catch!

Reply:
So I implemented the forwarding of the filters for the --exclude option and added a test case, but I think it is not meaningful to forward any usage-message. The user is using the diff-command and will get an appropriate message, wheres it would not be meaningful to print any usage message for the other command 'show' when using the 'diff' command. Would you agree?